Libraries/LibGUI/GTextEditor.cpp

```diff
@@ -573,6 +574,37 @@ void GTextEditor::move_selected_lines_down()
     update();
 }
 
+void GTextEditor::sort_selected_lines()
+{
+    if (is_readonly())
+        return;
+
+    if (has_selection()) {
+        int first_line;
+        int last_line;
+        get_selection_line_boundaries(first_line, last_line);
+
+        auto& lines = document().lines();
+
+        Vector<GTextDocumentLine> lines_to_sort;
+        for (int i = first_line; i <= last_line; i++) {
+            lines_to_sort.append(lines.at(i));
+        }
+
+        quick_sort(lines_to_sort.begin(), lines_to_sort.end(), [](auto& a, auto& b) {
```

Comment:
Could you use `quick_sort` directly on lines() here instead of creating a temporary `Vector<GTextDocumentLine>`?

Also, @rhin123: how do you think functionality like this should integrate with the undo stack?

Reply:
That's what I originally wanted to do as aforementioned, but I think I got caught up in the late hours of the night with unfamiliar syntax in an unfamiliar language. I think I got it now :),